test/functional/test_framework/wallet.py

```diff
@@ -73,7 +73,8 @@ def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_sp
 
         tx_info = from_node.testmempoolaccept([tx_hex])[0]
         self._utxos.append({'txid': tx_info['txid'], 'vout': 0, 'value': send_value})
-        from_node.sendrawtransaction(tx_hex)
+        if submit_tx:
```

Comment:
could this be something like `if not submit_tx: return tx` because that would help `mempool_limit.py` as well without much changes?

Reply:
You can also rebuild the `tx` from the `hex` field that this returns as-is I think
```python
from test_framework.messages import CTransaction, ToHex
hex = wallet.send_self_transfer(...)["hex"]
tx = FromHex(CTransaction(), hex)
```

Reply:
I see thanks :)